src/OpenTelemetry.Extensions.Enrichment/IsExternalInit.cs

```diff
@@ -0,0 +1,26 @@
+// <copyright file="IsExternalInit.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.ComponentModel;
+
+namespace System.Runtime.CompilerServices;
+
+/* This enables support for C# 9/10 records on older frameworks */
+
+[EditorBrowsable(EditorBrowsableState.Never)]
+internal static class IsExternalInit
```

Comment:
nit: Fine to keep this if you want but FYI you can also link into the project...

```xml
  <ItemGroup>
    <Compile Include="$(RepoRoot)\src\OpenTelemetry.Internal\IsExternalInit.cs" Link="Includes\IsExternalInit.cs" />
  </ItemGroup>
```

Reply:
Done. However, I have updated it a bit and made the class static because I was getting warnings like "this class is never instantiated". I hope this is fine with you

Reply:
As long as it compiles I'm good with it 😄